src/Microsoft.ML.AutoML/TrainerExtensions/SweepableParams.cs

```diff
@@ -172,5 +177,29 @@ public static IEnumerable<SweepableParam> BuildSymSgdLogisticRegressionParams()
                 new SweepableDiscreteParam("UpdateFrequency", new object[] { "<Auto>", 5, 20 })
             };
         }
+
+        /// <summary>
+        /// Gets the name of every hyperparameter swept across all trainers.
+        /// </summary>
+        public static ISet<string> GetAllSweepableParameterNames()
+        {
+            var sweepableParams = new List<SweepableParam>();
```

Comment:
is there a mechanism to enforce not to forget to update this list when new components are added?  looks to me this somehow needs to work off of a catalog and not have an explicit list at all

Reply:
no, there is not. I had the same concern. hopefully, a dev would click 'explore usages' of the existing methods in VS, and see all the usages, but this is not guaranteed, +1.
'LabelColumnName', 'ExampleWeightColumnName' etc are trainer properties here. We could use a blacklist instead, but I chose a whitelist to err hard on the side of not logging PII
would you be able to elaborate on your catalog idea in a concrete way? I would love a better solution to this, +1